### PR TITLE
[BUGFIX release] node 8

### PR DESCRIPTION
**Pull request by stefanpenner**
- [x] needs ember-cli-babel update (blocked on NPM not letting me release it...)
- [x] ember-try https://github.com/ember-cli/ember-try/pull/129
- [x] ember-disable-prototype-extensions (pending release -> https://github.com/ember-cli/ember-disable-prototype-extensions/issues/11)
- [x] qunitjs: https://github.com/qunitjs/qunit/pull/1184 (pending release, thanks @trentmwillis )
- [ ] not ok 1153 Acceptance: ember install installs addons via npm and runs generators
  Error: Command failed: npm install bower@^1.3.12
  npm WARN deprecated bower@1.8.0: ..psst! While Bower is maintained, we recommend Yarn and Webpack for *new* front-end projects! Yarn's advantage is security and reliability, and Webpack's is support for both CommonJS and AMD projects. Currently there's no migration path but we hope you'll help us figure out one.
  npm ERR! Maximum call stack size exceeded
  
  npm ERR! A complete log of this run can be found in:
  npm ERR!     /home/travis/.npm/_logs/2017-06-01T23_00_13_071Z-debug.log
  
      at Promise.all.then.arr (node_modules/execa/index.js:201:11)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:169:7)

### CODE REVIEW

**Comment by rwjblue** [comment]
@stefanpenner - Needs to target release branch

File: package.json [hunk 1]
```diff
@@ -147,7 +147,7 @@
     "yuidocjs": "0.10.2"
   },
   "engines": {
-    "node": "^4.5 || 6.* || 7.*"
+    "node": ">= ^4.5"
```
Comment (rwjblue) [inline package.json]: This is incorrect, cannot use both `>=` and `^` together.

Comment (stefanpenner) [inline package.json]: GH is trolling me, this isn't local...

Comment (stefanpenner) [inline package.json]: wrong repo..

**Comment by stefanpenner** [comment]
Likely also needs ember-cli-babel bump (but there is a cycle, as that needs this bump...)

Disabling node 8 on ember-cli-babel CI for now, I'll get that release out then circle back to this one.

**Comment by stefanpenner** [comment]
next up:
```

  1) platform-checker checkIsDeprecated:

      versions above the range are not deprecated
      + expected - actual

      -true
      +false

      at Context.<anonymous> (tests/unit/utilities/platform-checker-test.js:11:77)

  2) platform-checker checkIsValid:

      versions above the range are valid
      + expected - actual

      -false
      +true

      at Context.<anonymous> (tests/unit/utilities/platform-checker-test.js:18:72)


```

**Comment by stefanpenner** [comment]
1 job failed, seemed related to network. Re-running (rest past)

**Comment by stefanpenner** [comment]
more fixes, more togo..

**Comment by stefanpenner** [comment]
npm v5 infinite loop is happening here: https://github.com/npm/npm/blob/latest/lib/install.js#L363-L381

Seems to be related to:

```
ember-cli/tmp/<something>/dummy-app/node_modules/<ember-cli symlink back to the top folder>
```

**Comment by stefanpenner** [comment]
more progress debugging root cause I believe is that https://github.com/npm/read-package-tree although I believe intends to return a tree can return a cyclic graph.

Going to write tests and stuff for that later. I need to go home and eat food now.

**Comment by stefanpenner** [comment]
issue for npm v5: https://github.com/npm/npm/issues/16994

File: .travis.yml [hunk 2]
```diff
@@ -5,6 +5,7 @@ node_js:
   - "4"
   - "6"
   - "7"
+  - "8"
```
Comment (Turbo87) [inline .travis.yml]: do we still need to test against 7 if we test against 8 now?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,31 @@
...
+- [#7023](https://github.com/ember-cli/ember-cli/pull/7023) Allow broccoli-babel-transpiler to float with SemVer. [@rwjblue](https://github.com/rwjblue)
+- [#7028](https://github.com/ember-cli/ember-cli/pull/7028) add yarn missing default comment [@kellyselden](https://github.com/kellyselden)
+- [#7036](https://github.com/ember-cli/ember-cli/pull/7036) Ensure `lintTree` results cannot clobber tests. [@rwjblue](https://github.com/rwjblue)
+- [#7049](https://github.com/ember-cli/ember-cli/pull/7049) Prevent warnings from broccoli-babel-transpiler. [@rwjblue](https://github.com/rwjblue)
+- [#7051](https://github.com/ember-cli/ember-cli/pull/7051) Corrected a typo in Windows elevation test error message. [@Turbo87](https://github.com/Turbo87)
+- [#7052](https://github.com/ember-cli/ember-cli/pull/7052) Pass only package to npm uninstall task that exist [@Turbo87](https://github.com/Turbo87)
+
+Thank you to all who took the time to contribute!
+
+
```
Comment (Turbo87) [inline CHANGELOG.md]: ehm... why is there a release embedded in this PR?!

File: blueprints/app/files/package.json [hunk 4]
```diff
@@ -38,7 +38,7 @@
     "loader.js": "^4.2.3"
   },
   "engines": {
-    "node": ">= 4"
+    "node": "^4.5 || 6.* || >= 7.*.*.*.*.*.*.*.*"
```
Comment (Turbo87) [inline blueprints/app/files/package.json]: `7.*.*.*.*.*.*.*.*`?

File: package.json [hunk 5]
```diff
@@ -60,7 +60,8 @@
     "compression": "^1.4.4",
     "configstore": "^3.0.0",
     "console-ui": "^1.0.2",
-    "core-object": "^3.0.0",
+    "core-object": "^3.1.3",
+    "dag-map": "^2.0.2",
```
Comment (Turbo87) [inline package.json]: where is this used?

File: package.json [hunk 6]
```diff
@@ -141,12 +142,12 @@
...
     "nock": "^9.0.6",
     "supertest": "^3.0.0",
-    "testdouble": "^2.0.1",
+    "testdouble": "^2.1.2",
     "yuidoc-ember-cli-theme": "^1.0.0",
     "yuidocjs": "0.10.2"
   },
   "engines": {
-    "node": "^4.5 || 6.* || 7.*"
+    "node": "^4.5 || 6.* || >= 7.*"
```
Comment (Turbo87) [inline package.json]: are you sure that we are compatible with Node 42.3?

File: .travis.yml (hunk 2 above)
Comment (rwjblue) [inline .travis.yml]: yes, we should not immediately show warnings to folks on node@7 (in a patch release)

We should remove node@7 on master after the next beta branch point

File: blueprints/app/files/package.json (hunk 4 above)
Comment (rwjblue) [inline blueprints/app/files/package.json]: O_o

File: package.json (hunk 5 above)
Comment (rwjblue) [inline package.json]: dag-map is used for addon discovery/ordering

Comment (Turbo87) [inline package.json]: and it was a missing dependency previously? or why is this being added here? seems unrelated to the Node 8 update

File: CHANGELOG.md (hunk 3 above)
Comment (stefanpenner) [inline CHANGELOG.md]: oops

File: blueprints/app/files/package.json (hunk 4 above)
Comment (stefanpenner) [inline blueprints/app/files/package.json]: i lost my mind

File: package.json (hunk 6 above)
Comment (stefanpenner) [inline package.json]: @Turbo87 they can try, and we will warn.

**Comment by ro0gr** [comment]
> npm ERR! Maximum call stack size exceeded

I wonder if `npm@5` is really blocking here. I thought the intention is to use `yarn` for the test suite?

**Comment by kanongil** [comment]
I would say that https://github.com/quaertym/ember-cli-dependency-checker/issues/76 needs to be fixed as well. Maybe the one here: https://github.com/quaertym/ember-cli-dependency-checker/pull/77?

**Comment by rwjblue** [comment]
@kanongil - merged and released in ember-cli-dependency-checker@2.0.1

**Comment by stefanpenner** [comment]
The test failure doesn't exist on master, I wonder if release just got into a funky state for that 1 test on windows...

**Comment by stefanpenner** [comment]
paired with @rwjblue, we have decided to merge this. The one failing test on release is a funky NPM issue that crept up....

**Comment by stefanpenner** [comment]
@twokul can we cut a 2.13.x which includes these commits?

**Comment by twokul** [comment]
@stefanpenner sure thing

**Comment by twokul** [comment]
@stefanpenner published `2.13.3` https://github.com/ember-cli/ember-cli/releases/tag/v2.13.3
